tests/reference/cpp-doconcurrentloop_01-4e9f274.stdout

```diff
@@ -28,11 +28,11 @@ void main0()
     Kokkos::View<float[10000]> c("c");
     int nsize;
     float scalar;
-    scalar = 10;
+    scalar = (float)(10);
     nsize = 1234;
     Kokkos::parallel_for(Kokkos::RangePolicy<Kokkos::DefaultExecutionSpace>(1, nsize - 1+1), KOKKOS_LAMBDA(const long i) {
-        a[i-1] = 5;
-        b[i-1] = 5;
+        a[i-1] = (float)(5);
+        b[i-1] = (float)(5);
```

Comment:
This however is correct, as the integer gets upgraded to a float, because the LHS is an array, so things get implicitly cast.